### ISSUE TITLE
Starting in Landscape Orientation Issue

### ISSUE THREAD

**Issue by NathanaelA** [NativeScript/NativeScript#1656]
### Did you verify this is a real problem by searching
yes

### Tell us about the problem
If you start the application in Landscape mode, then rotate the screen the app does NOT recognize the very first rotation.  Starting Portrait ->Landscape works fine. 

Issue is located at: https://github.com/NativeScript/NativeScript/blob/master/application/application.android.ts#L323

I debugged this a bit, this line appears to return **1** at startup no matter WHAT orientation the device is.  So when we enter https://github.com/NativeScript/NativeScript/blob/master/application/application.android.ts#L329 on the rotation to portrait = 1; so it falls out of the check without doing anything and so the very first Landscape -> Portrait fails to call my event listener.

### Which platform(s) does your issue occur on?
Android (Geny and Real s5 device)

### Please provide the following version numbers that your issue occurs with:
Runtime: 1.62
Core: 1.62
TNS: 1.61
Android OS: 4.4

### Please tell us how to recreate the issue in as much detail as possible.
1. Start up App; Rotate; Quit and terminate app.  (Don't want it still running, but you need Android to know the app support landscape mode.
2. Startup App (while still in Landscape)
3. Rotate screen

With the demo code below the Console SHOULD print out "Test Orientation called" when you rotate from Landscape to Portrait.    

### Is there code involved? If so, please share the minimal amount of code needed to recreate the problem.
```xml
<Page loaded="pageLoaded" unloaded="pageUnloaded">
  <StackLayout>
   <Label id="statusLabel" text="Hi" />
  </StackLayout>
</Page>
```

```js
var application = require('application');

exports.pageLoaded = function() {
    application.on(application.orientationChangedEvent, testOrient);
};

exports.pageUnloaded = function() {
    application.off(application.orientationChangedEvent, testOrient);
};

function testOrient() {
    console.log("Test Orientation called");
}
```

**Comment by vbresults** [comment]
I can't get the orientation changed event to work at all if I don't pass the 3rd page parameter to application.on, not a showstopper but thought I'd mention it if you are using the above code.

**Comment by NathanaelA** [comment]
The third parameter is the "this".    It isn't needed in the case the routine is not part of a class.   The code above works fine in all situations except if the app is started in Landscape mode and so the very first "landscape" to "portrait" mode switch.   I traced down the reason why, but I'm not sure the purpose for the caching the initial orientation check 

  I'm currently "modifying" my Core-modules for this product and defaulting the value to 0.  That way all the rotations works...

**Comment by NathanaelA** [comment]
Not sure this is a "proper" fix (more like a working hack); but if you toss this (https://github.com/NativeScript/NativeScript/blob/master/application/application.android.ts#L309) into a setTimeout:
```
setTimeout(function() {
  setupOrientationListener(androidApp);
},0);
```
It fixes it.  So I assume the issue is that it is initializing too soon or something.   If this is in a setTimeout the value for currentOrientation = 2 properly...

**Comment by vbresults** [comment]
The routine wasn't a part of the class -- it was in app.js, regular 4-liner

**Comment by NathanaelA** [comment]
Hmm, that is weird -- was that maybe an issue in v1.5?     I haven't passed the third parameter in any of my screens, but I only started using this feature in v1.6.

**Comment by vbresults** [comment]
Perhaps it was, that was when I tried it

**Comment by vjoao** [comment]
Found another bug related to the Landscape, dunno if I should create another thread for this.
When the app starts in landscape mode, the ActionBar title font size is smaller. When the phone is rotated, the size stays the same.

**Comment by mikeres0** [comment]
@vjoao I've seen this too, so I'd probably create another thread :)

**Comment by vbresults** [comment]
The built in orientation detection code is wonky for iOS at least, I had to write a module to work around this.

```
var orientation = require("./orientation-common"),
	enums = require("ui/enums");

orientation.get = function(interfaceOrientation) {
	if (typeof interfaceOrientation === "undefined") {
		interfaceOrientation = UIDevice.currentDevice().orientation;
	}

	switch (interfaceOrientation) {
		case UIDeviceOrientation.UIDeviceOrientationLandscapeRight:
		case UIDeviceOrientation.UIDeviceOrientationLandscapeLeft:
		case UIDeviceOrientation.UIDeviceOrientationPortraitUpsideDown:
			return enums.DeviceOrientation.landscape;
		case UIDeviceOrientation.UIDeviceOrientationPortrait:
			return enums.DeviceOrientation.portrait;
		case UIDeviceOrientation.UIDeviceOrientationFaceUp:
		case UIDeviceOrientation.UIDeviceOrientationFaceDown:
			return module.exports.get(UIApplication.sharedApplication().statusBarOrientation);
	}
};

module.exports = orientation;
```

**Comment by enchev** [comment]
Hey folks,

Do you still have the same issue?

**Comment by mikeres0** [comment]
@enchev I haven't used NS for a while as I've got side tracked, so I won't be able to test this! :-1:

**Comment by NathanaelA** [comment]
@enchev -- This has been fixed in at least 2.1, I just retested it; and everything seems to be good; and the old code no longer looks to be present in the startup function, so you are now only using the change orientation event without trying to pre-populate a (wrong) cached value...
